src/appleseed/renderer/modeling/environmentedf/nishita93environmentedf.cpp

```diff
@@ -0,0 +1,494 @@
+            float                       m_sun_angular_diameter;             // rays with angle (rad) +- sun_angular_diameter will return sun radiance directly
+            bool                        m_precompute;                       // use 2D lookup table to precompute sky optical depths
+
+            // Fills a 3D precomputation table with optical depths value along the sun direction.
+            void sky_precomputations() {
+                nishita::precompute_mie_g(m_haze);
+                nishita::precompute_shells();
+                if (m_precompute)
+                    nishita::precompute_optical_depths(sun_dir, m_air_molecule_density, m_dust_molecule_density, m_ozone_molecule_density);
+            }
+
+            // Compute the sky radiance along a given direction.
+            void compute_sky_radiance(
+                const ShadingContext&   shading_context,
+                const Vector3f&         outgoing,
+                RegularSpectrum31f&     radiance) const
+            {
+
+                // Position of the camera at least 1.2 meters above earth radius to avoid numerical errors.
+                Vector3f camera_position = Vector3f(0.0f, earth_radius + 1.2f + m_elevation, 0.0f);
+                Ray3f ray = Ray3f(camera_position, outgoing);
+
+                // If the outoing vector points to the sun, return suns spectrum.
+                float sun_angular_radius = m_sun_angular_diameter / 2.0f;
+                float is_sun = norm(outgoing - sun_dir) < sun_angular_radius;
+                if (is_sun) {
+                    const bool sun_hit = nishita::sun_disk(
+                        ray,
+                        m_air_molecule_density,     // air molecule density (Rayleigh scattering)
+                        m_dust_molecule_density,    // dust molecule density (Mie scattering)
+                        m_ozone_molecule_density,   // ozone molecule density (Ozone scattering)
+                        sun_angular_radius,
+                        radiance
+                    );
+                    if (sun_hit)
+                        return;
+                }
+
+                // Compute the final sky radiance.
+                nishita::single_scattering(
+                    ray,
+                    sun_dir,                    // sun direction
+                    m_air_molecule_density,     // air molecule density (Rayleigh scattering)
+                    m_dust_molecule_density,    // dust molecule density (Mie scattering)
+                    m_ozone_molecule_density,   // ozone molecule density (Ozone scattering)
+                    m_precompute,               // use precomputed lookup table
+                    radiance
+                );
+                radiance *=
+                    m_uniform_values.m_sun_intensity_multiplier     // multiply sun intensity
+                    * 1.5f;         // since nishita93 underestimates radiance by 1/3 according to Bruneton
```

Comment:
Could you link the source paper for this statement? Just in case, someone wants to check it out at a later time.
Also, please aling these comments (you can also just make it one standalone comment).